nightly (1.9) tests failing

https://github.com/JuliaLang/PackageCompiler.jl/runs/7175757050?check_suite_focus=true#step:4:341
```
/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c: In function ‘main’:
/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c:107:5: warning: implicit declaration of function ‘jl_set_global’; did you mean ‘jl_get_global’? [-Wimplicit-function-declaration]
  107 |     jl_set_global(jl_base_module, jl_symbol("PROGRAM_FILE"), firstarg);
      |     ^~~~~~~~~~~~~
      |     jl_get_global
```

It appears `jl_set_global` was removed in https://github.com/JuliaLang/julia/pull/44753: @vtjnash is this considered a breaking API change?

--- a/docs/src/devdocs/binaries_part_2.md
+++ b/docs/src/devdocs/binaries_part_2.md
@@ -175,8 +175,12 @@ int main(int argc, char *argv[])
     jl_set_ARGS(argc, argv);
 
     // Set PROGRAM_FILE to argv[0].
-    jl_set_global(jl_base_module,
-        jl_symbol("PROGRAM_FILE"), (jl_value_t*)jl_cstr_to_string(argv[0]));
+#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 9
+    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, jl_symbol("PROGRAM_FILE"), 1);
+    jl_checked_assignment(bp, (jl_value_t*)jl_cstr_to_string(argv[0]));
+#else
+    jl_set_global(jl_base_module, jl_symbol("PROGRAM_FILE"), (jl_value_t*)jl_cstr_to_string(argv[0]));    
+#endif
 
     // Set Base.ARGS to `String[ unsafe_string(argv[i]) for i = 1:argc ]`
     jl_array_t *ARGS = (jl_array_t*)jl_get_global(jl_base_module, jl_symbol("ARGS"));

--- a/src/embedding_wrapper.c
+++ b/src/embedding_wrapper.c
@@ -104,7 +104,12 @@ int main(int argc, char *argv[]) {
     checked_eval_string("append!(empty!(Base.ARGS), Core.ARGS)");
     jl_value_t *firstarg = checked_eval_string("popfirst!(ARGS)");
     JL_GC_PUSH1(&firstarg);
-    jl_set_global(jl_base_module, jl_symbol("PROGRAM_FILE"), firstarg);
+#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 9
+    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, jl_symbol("PROGRAM_FILE"), 1);
+    jl_checked_assignment(bp, firstarg);
+#else
+    jl_set_global(jl_base_module, jl_symbol("PROGRAM_FILE"), firstarg);    
+#endif
     JL_GC_POP();
 
     // call the work function, and get back a value

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -132,7 +132,8 @@ end
 
             io = IOBuffer()
             p = run(pipeline(ignorestatus(`$(app_path("Undefined"))`), stderr=io;))
-            @test occursin("UndefVarError: undefined not defined", String(take!(io)))
+            str = String(take!(io))
+            @test all(occursin(str), ["UndefVarError:", "undefined", "not defined"])
             @test p.exitcode == 1
         end
     end

